site/it/tutorials/customization/basics.ipynb

```diff
@@ -1,480 +1,480 @@
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 0,
+      "metadata": {
+        "cellView": "code",
+        "colab": {},
+        "colab_type": "code",
+        "id": "ngUe237Wt48W"
+      },
+      "outputs": [],
+      "source": [
+        "print(tf.add(1, 2))\n",
+        "print(tf.add([1, 2], [3, 4]))\n",
+        "print(tf.square(5))\n",
+        "print(tf.reduce_sum([1, 2, 3]))\n",
+        "\n",
+        "# Operator overloading is also supported\n",
+        "print(tf.square(2) + tf.square(3))"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "IDY4WsYRhP81"
+      },
+      "source": [
+        "Ogni oggetto `tf.Tensor` ha una forma e un tipo di dato:"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 0,
+      "metadata": {
+        "colab": {},
+        "colab_type": "code",
+        "id": "srYWH1MdJNG7"
+      },
+      "outputs": [],
+      "source": [
+        "x = tf.matmul([[1]], [[2, 3]])\n",
+        "print(x)\n",
+        "print(x.shape)\n",
+        "print(x.dtype)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "eBPw8e8vrsom"
+      },
+      "source": [
+        "Le differenze più ovvie tra gli array NumPy e i `tf.Tensor` sono:\n",
+        "\n",
+        "1. I tensori possono essere supportati da acceleratori di memoria (come GPU, TPU).\n",
+        "2. I tensori sono immutabili."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "Dwi1tdW3JBw6"
+      },
+      "source": [
+        "### Compatibilità con NumPy \n",
+        "\n",
+        "La conversione tra oggetti `tf.Tensor` di TensorFlow e un `ndarray` di NumPy è facile:\n",
+        "\n",
+        "* Le operazioni di TensorFlow convertono automaticamente gli ndarrays di NumPy a tensori.\n",
+        "* Le operazioni di NumPy convertono automaticamente i tensors a ndarrays di NumPy.\rappresentazione di memoria, se possible",
+        "\n",
+        "I tensori sono convertiti esplicitamente ad ndarrays di NumPy usando il loro metodo. Queste conversioni sono tipicamente economiche dal momento che l'array e `tf.Tensor` condividono la stessa rappresentazione di memoria, se possible. Tuttavia, condividere la rappresentazione di memoria sottostante non è sempre possibile dal momento che `tf.Tensor` può essere hostata nella memoria GPU mentre gli array NumPy sono sempre supportati dalla memoria, e la conversazione coinvolge una copia dalla GPU e mentre gli array numPy sono sempre protetti dalla memoria."
```

Comment:
sono convertiti esplicitamente ad ndarrays di NumPy --> sono convertiti esplicitamente in ndarrays di NumPy
economiche --> efficienti
condividono la stessa rappresentazione di memoria, se possible. -->  condividono, se possible, la stessa rappresentazione di memoria.
può essere hostata nella memoria GPU mentre gli array NumPy sono sempre supportati dalla memoria, -->
può essere allocata nella memoria GPU mentre gli array NumPy sono sempre ospitati nella memoria centrale
e la conversazione coinvolge una copia dalla GPU e mentre gli array numPy sono sempre protetti dalla memoria. -->
e la conversione coinvolge una copia dalla GPU alla memoria centrale.

Reply:
All the proposed sentences are good to me, I agree with them